frontend/src/app/libs/common-components/components/tasks-creator/tasks-creator/tasks-creator.component.ts

```diff
@@ -0,0 +1,352 @@
+  }
+
+  get singleAnswersControl(): FormArray {
+    return this.taskCreatorControl?.get('answersArrayControl') as FormArray ?? null;
+  }
+
+  get textEditorQuestion(): string {
+    return this.taskCreatorControl.get('textEditorControl')?.value ?? null;
+  }
+
+  get correctSingleAnswer(): number {
+    return parseInt(this.taskCreatorControl.get('correctSingleAnswerControl')?.value, 10) ?? null;
+  }
+
+  get multiAnswerControl(): FormArray {
+    return this.taskCreatorControl.get('multiAnswersArrayControl') as FormArray ?? null;
+  }
+
+  get isListEmpty(): boolean {
+    return !Boolean(this.selectedTask);
+  }
+
+  set isMentorCheckSelected(value: boolean) {
+    this.taskCreatorControl.get('mentorCheckControl')?.setValue(value);
+  }
+
+  set maxScore(value: number) {
+    this.taskCreatorControl.get('maxScoreControl')?.setValue(value);
+  }
+
+  set taskNameControl(value: string) {
+    this.taskCreatorControl.get('taskNameControl')?.setValue(value ?? '');
+  }
+
+  set textEditorQuestion(value: string) {
+    this.taskCreatorControl.get('textEditorControl')?.setValue(value ?? '');
+  }
+
+  set selectedTaskType(value: TasksTypes) {
+    this.taskCreatorControl.controls['selectedTaskTypeControl'].setValue(value);
+  }
+
+  public onOpenTaskTypesSelector(isOpen: boolean): void {
+    this.isTaskTypesSelectorOpened = isOpen;
+  }
+
+  public enableTaskNameEditMode(): void {
+    this.taskCreatorControl.addControl('taskNameControl', new FormControl());
+    this.taskNameControl = this.taskName;
+    this.isTaskNameEditMode = true;
+  }
+
+  public rebuildFormControls(tasksTypes: TasksTypes): void {
+    switch (tasksTypes) {
+      case TasksTypes.single:
+        this.removeControlsOfTaskType(TasksTypes.single);
+        this.taskCreatorControl.addControl('correctSingleAnswerControl', new FormControl());
+        this.taskCreatorControl.addControl('answersArrayControl', this.formBuilder.array([]));
+        break;
+
+      case TasksTypes.multiple:
+        this.removeControlsOfTaskType(TasksTypes.multiple);
+        this.taskCreatorControl.addControl('multiAnswersArrayControl', this.formBuilder.array([]));
+        break;
+
+      case TasksTypes.coding:
+        this.removeControlsOfTaskType(TasksTypes.coding);
+        this.taskCreatorControl.addControl('codeEditorControl', new FormControl());
+        break;
+
+      case TasksTypes.text:
+        this.removeControlsOfTaskType(TasksTypes.text);
+        break;
+
+      default:
+    }
+  }
+
+  public disableTaskNameEditMode(): void {
+    this.taskName = this.taskNameFromControl;
+    this.taskCreatorControl.removeControl('taskNameControl');
+    this.isTaskNameEditMode = false;
+  }
+
+  public ngOnInit(): void {
+    this.rebuildFormControls(this.selectedTaskType);
+
+    this.taskCreatorControl.valueChanges.subscribe(
+      () => {
+        if (this.selectedTaskType === TasksTypes.coding && this.isMentorCheckSelected === false) {
```

Comment:
`this.isMentorCheckSelected === false` the same as `!this.isMentorCheckSelected`